### ISSUE TITLE
Support Java 11+ for SPF

### ISSUE THREAD

**Issue by wikumC** [SymbolicPathFinder/jpf-symbc#47]
I went through your GSOC project ideas page and found out that you are looking for a prospective student to add java 11+ support for jpf-symbc.  I built [jpf-core](https://github.com/javapathfinder/jpf-core) and [jpf-symbc](https://github.com/SymbolicPathFinder/jpf-symbc) using java 8 and test run some of the examples given. Then I tried to build jpf-symbc using Java 11. First I build jpf-core with Java 11 as jpf-symbc requires to have jpf-core's class files on the classpath. But building jpf-symbc using Java 11, results in several build errors and deprecation warnings.

The main issue I faced is when compiling Java module classes such as [Bufferedimage.java](https://github.com/SymbolicPathFinder/jpf-symbc/blob/67165d4119a39cc8280b5eea47eb97418db38d18/src/classes/java/awt/image/BufferedImage.java), which caused build errors like the following:

```
error: package exists in another module: java.desktop
[javac] package java.awt.image;

error: cannot find symbol
[javac]     public BufferedImage(int width, int height, int imageType, IndexColorModel cm) {
[javac]                                                                ^
```
I was able to fix these split package errors and successfully build the project.
 Will you be accepting PRs to add support for Java 11? If so I'd be happy to make a PR from my fork.

**Comment by corinus** [comment]
Yes please

**Comment by corinus** [comment]
Hi Yes it would be great.
The BufferedImage is just an example. I can remove it.

**Comment by wikumC** [comment]
It seems there is only the master branch in this repo. For the movement, I think it will better to create my PR against a new [Java 11 specific branch](https://github.com/javapathfinder/jpf-core/tree/java-10-gradle) like in jpf-core. Maybe we could merge that branch to the master once we fully support Java 11. 

Would you be able to create a new branch?

**Comment by corinus** [comment]
Hi May I ask if you applied to GSoC? Please send me an email directly! Thanks. Corina

**Comment by wikumC** [comment]
> Hi May I ask if you applied to GSoC? Please send me an email directly! Thanks. Corina

I sent you an email.

**Comment by corinus** [comment]
I did not receive

**Comment by wikumC** [comment]
> I did not receive



> I did not receive

**Comment by wikumC** [comment]
> I did not receive

I sent an email to `Corina.S.Pasareanu(at)nasa.gov` on March 07. Can you please recheck? I resent the same email today just make sure.

**Comment by ranjsa** [comment]
Hi 
@corinus are you mentoring for SymbolicPathFinder 
I am pursuing my computer science engineering, and I am very good with Java and Algorithms.

I want to contribute in Support of Java 11 for Spf 
Since the projects are announced I have solved some issue in jpf-core and getting grip for both the projects jpf -core and spf.

I want to contact with a mentor for discussing further learnings and questions that have arisen during exploring these projects .

**Comment by corinus** [comment]
Hi Yes I am mentoring for GSoC. Please send a message to corina.pasareanu@west.cmu.edu not to the NASA email. Thank you.

**Comment by ranjsa** [comment]
@corinus  
Okay  
Thank you. 
I will get back to you soon

**Comment by ThanoshanMV** [comment]
Hi, does this issue still open?

**Comment by ThanoshanMV** [comment]
Hi, is this issue still open?

**Comment by aosen-xiong** [comment]
Looks like the issue is still open and GSOC 2024 has a call for this. I am wondering what is difference between https://github.com/SymbolicPathFinder/jpf-symbc and https://github.com/javapathfinder/jpf-symbc. And also which will be maintained in the future.

**Comment by sohah** [comment]
Yes, I am too confirming that the issue is still open, and yes it is part of GSoC 2024 project ideas. 
Not sure about the multiple repos though. @yannicnoller, can you please clarify/answer Ao-senXiong's question?

**Comment by yannicnoller** [comment]
Hi @Ao-senXiong @sohah, I was not aware of the fork on the javapathfinder account. Please focus on the https://github.com/SymbolicPathFinder/jpf-symbc.

... and yes, the issue is still open. Feel free to join our Discord server to discuss more: https://discord.gg/sX4YZUVHK7
